src/ColumnNumberAwareTrait.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Dhii\Parser\Tokenizer;
+
+use InvalidArgumentException;
+use Exception as RootException;
+
+/**
+ * Functionality that allows storage and retrieval of a column number.
+ *
+ * @since [*next-version*]
+ */
+trait ColumnNumberAwareTrait
+{
+    /**
+     * @since [*next-version*]
+     *
+     * @var ColumnNumberInterface|null
+     */
+    protected $columnNumber;
+
+    /**
+     * Retrieves the column number associated with this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @return int|null
```

Comment:
Here and everywhere else, include a description for the return value. It may be overly simple for something like `TokenizerAwareTrait`, but for cases like this it would be helpful to specify the nature of the `int`, such as being a positive 1-based number.